### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by Paskab012**
## Description of changes made ♻️
- Set up the repository and tools.
- Create your models according to the ERD diagram. Don't forget about validations and tests.
- Implement the authentication and authorization.
- Create the page to add a category.
- Create the home page.
- Create the page to add a transaction.
- Create the transactions page.
- Make sure your app is tested adequately.
- Deploy the project, and test for final details.
- Record a video for your project.
- Create a good README and PR description.

#### Type of change

> **Please select the relevant options**

- [ ] New feature (non-breaking changes which add the new functionalities)
- [ ] Bug fix (non-breaking changes which fix an issue)
- [ ] Chore (non-breaking changes)
- [ ] Refactoring (non-breaking changes)
- [ ] Documentation (added or improved the documentation)
- [ ] Breaking change (fix or a feature that would cause existing feature not to work as expected)
- [ ] This change requires a documentation update

#### How Has This Been Tested?

> **Please describe the tests that you ran to verify your changes**

- [x] End to end
- [x] Manual

#### How should this be manually tested?

> **Provide the steps to manually test the functionalities implemented in this PR**

- After cloning the [repo](https://github.com/Paskab012/rails_budget_app.git)
- Check out this branch by `branch here`
- Run `yarn or npm install to install the project dependencies
- Run `rails s` to start the development server
- bundle install

#### Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have ensured coding conventions for my code prior to submission
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove the effectiveness of my work
- [ ] New and existing tests pass locally with my changes

#### Screenshots

|        BEFORE                  |                     AFTER       |
| :----------------------------: | :-----------------------------: |
|                                | Note: You can even provide a    |
|                                | screenshot                      |

### CODE REVIEW

File: spec/views/transaction_index_spec.rb [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  it 'should display category name the transaction belongs to' do
+    expect(page).to have_content('Basketball')
+  end
+
+  it 'should display the transaction amount' do
+    expect(page).to have_content('0$')
+  end
+
+  it 'page path to be transaction path' do
+    expect(page).to have_current_path category_transactions_path(@category.id)
```
Comment (RolandM99) [inline spec/views/transaction_index_spec.rb]: - [x] Some of your tests failed, Kindly ensure that all your tests pass successfully.
image

**Review by RolandM99 — CHANGES_REQUESTED** [review]
Hello @Paskab012, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

### To highlight 💯 
- Good commit message ✔️ 
- No Linter errors ✔️ 
- Follow Github flow ✔️ 

## Required Changes ♻️
_Check the comments under the review._

- [x] Make sure that you deployed well your app. For now, there is an error that appears when we click on the signup button. Kindly, fix it.
image
 

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me on @Rolandm99 in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/views/transaction_index_spec.rb (hunk 1 above)
Comment (Paskab012) [reply spec/views/transaction_index_spec.rb]: Thanks @RolandM99 for your feedback. Let me look at a way to resolve the issue

Comment (Whoistolu) [reply spec/views/transaction_index_spec.rb]: Well resolved 👍🏽

**Review by Whoistolu — APPROVED** [review]
### Hi @Paskab012 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

# STATUS: APPROVED 🟢
**Congratulations! 🎉** 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @whoistolu in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**